libs/user-facing-errors/src/lib.rs

```diff
@@ -1,4 +1,6 @@
-#![deny(warnings, rust_2018_idioms)]
+#![deny(unsafe_code, warnings, rust_2018_idioms)]
+#![allow(clippy::derive_partial_eq_without_eq)]
```

Comment:
Sonatype hasn't upgraded to rust 1.63.0 yet, this is just wrong.

Reply:
Is that not something we define and they should use from our lockfile?

Reply:
the rust version is (by design) not persisted in Cargo.lock or anywhere else in the rust code and cargo config

Reply:
Ah ok. So Sonatype can not know that we expect to run with that version.

Reply:
Nope — we just tend to use the latest version, and we haven't persisted that anywhere except the nix shell. We could have a `rust_toolchain` file from rustup that specifies it, though, we just don't.

Reply:
Not even in the nix shell, in the nix flake lockfile (implicitly by virtue of being pinned to a particular version of rust_overlay).

Reply:
The GH action that runs clippy is always on latest stable, as far as I can tell, so it started failing this morning (since there was the 1.63 release yesterday afternoon).